src/utils/displayTo.ts

```diff
@@ -1,33 +1,88 @@
 import { css, ThemedCssFunction } from './styled';
-import { ScreenSize, Breakpoints } from '../common';
+import { ScreenSize, Breakpoints, ScreenSizeList } from '../common';
 import { breakpoints } from '../themes';
 
-function isBreakpoint(breakpoints: Breakpoints, name: keyof Breakpoints) {
-  const value = breakpoints && breakpoints[name];
-  return typeof value === 'number' && value > 0;
+export function getMediaQueries(breakpoints: Breakpoints) {
+  return ScreenSizeList.map(x => ({ screen: x, breakpoints: getScreenSizeBreakpoints(x, breakpoints) })).reduce(
+    (acc, next) => (next.breakpoints ? { ...acc, [next.screen]: formatQuery(next.breakpoints) } : acc),
+    {},
+  );
 }
 
-export function getMediaQueries(breakpoints: Breakpoints) {
-  if (!isBreakpoint(breakpoints, 'medium') || !isBreakpoint(breakpoints, 'large')) {
-    throw new Error('Invaild breakpoints');
+// create max-width media query including screen size
+export function displayUpTo(screen: ScreenSize) {
+  const screenBreakpoints = getScreenSizeBreakpoints(screen, breakpoints);
+  if (!screenBreakpoints) {
+    throw new Error('Invalid screen size');
   }
+  const query = formatQuery({ next: screenBreakpoints.next });
+  return formatMedia(query);
+}
 
-  return {
-    small: `(max-width: ${breakpoints.medium - 1}px)`,
-    smallAndMedium: `(max-width: ${breakpoints.large - 1}px)`,
-    medium: `(min-width: ${breakpoints.medium}px) and (max-width: ${breakpoints.large - 1}px)`,
-    mediumAndLarge: `(min-width: ${breakpoints.medium}px)`,
-    large: `(min-width: ${breakpoints.large}px)`,
-  };
+// create min-width media query including screen size
+export function displayFrom(screen: ScreenSize) {
+  const screenBreakpoints = getScreenSizeBreakpoints(screen, breakpoints);
+  if (!screenBreakpoints) {
+    throw new Error('Invalid screen size');
+  }
+  const query = formatQuery({ prev: screenBreakpoints.prev });
+  return formatMedia(query);
 }
 
 export function displayTo(screen: ScreenSize | string): ThemedCssFunction<any> {
```

Comment:
Are we going to deprecate this function as well?

Reply:
No